library/pkparse.c

```diff
@@ -975,11 +1034,21 @@ static int pk_parse_key_sec1_der(mbedtls_ecp_keypair *eck,
         }
     }
 
-    if (!pubkey_done &&
-        (ret = mbedtls_ecp_mul(&eck->grp, &eck->Q, &eck->d, &eck->grp.G,
-                               f_rng, p_rng)) != 0) {
-        mbedtls_ecp_keypair_free(eck);
-        return MBEDTLS_ERROR_ADD(MBEDTLS_ERR_PK_KEY_INVALID_FORMAT, ret);
+    if (!pubkey_done) {
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+        (void) f_rng;
+        (void) p_rng;
+        if ((ret = pk_derive_public_key(&eck->grp, &eck->Q, &eck->d)) != 0) {
+            mbedtls_ecp_keypair_free(eck);
+            return MBEDTLS_ERROR_ADD(MBEDTLS_ERR_PK_KEY_INVALID_FORMAT, ret);
```

Comment:
Ah, sorry, should have caught this earlier, but `ERROR_ADD` only works between a high-level module (such as PK) and a low-level module (such as ECP) - see `error.h` for the scheme. Here `pk_derive_public_key()` potentially return PK errors (from `psa_pk_status_to_mbedtls()` and we shouldn't add two PK errors together.

I think just `return ret;` would be OK here.

Reply:
Sorry I should have checked that as well. I'll fix it!